src/server/enterprise/server/api_server.go

```diff
@@ -330,3 +343,192 @@ func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (
 
 	return &ec.DeactivateResponse{}, nil
 }
+
+func (a *apiServer) Pause(ctx context.Context, req *ec.PauseRequest) (resp *ec.PauseResponse, retErr error) {
```

Comment:
We should account for this API call being called on a cluster running in `enterprise` mode. I think right now we would transition such a cluster through `enterprise -> paused -> full`. If it's easiest, I think preventing pausing of an enterprise mode cluster would be fine for now

Reply:
I think with the ConfigMap approach the enterprise server is safe, as it gets started with a verbatim `--mode=enterprise`, with no reference to the ConfigMap value.

It does make sense to generate an error when trying to pause an enterprise server, though. Adding that …

Reply:
I think it may still be worth parameterizing it such that when you "unpause", the previous mode is reset as opposed to the hard-coded "full" mode. 

The case I have in mind, is that pretty soon we'll want to run pachd in a mode where the identity gRPC service is turned off (in the case where it's relying on the identity service of the enterprise server). Perhaps there are other ways of configuring it, but this could naturally be implemented with a corresponding mode which would be different than full.

Reply:
The previous mode could be stored as another value in the ConfigMap, then used to replace `MODE` on an unpause. What do you think of that as a solution?

Reply:
This is implemented now — let me know what you think.

Reply:
looks good!